goplay/src/main/java/be/tapped/goplay/content/ProgramRepo.kt

```diff
@@ -59,64 +48,33 @@ internal class HtmlFullProgramParser(private val jsoupParser: JsoupParser) {
             .flatMap {
                 Either.catch {
                     val programDataObject = Json.decodeFromString<JsonObject>(it)["data"]!!.jsonObject
-                    Json {
-                        isLenient = true
-                        ignoreUnknownKeys = true
-                    }.decodeFromJsonElement<Program>(programDataObject)
+                    jsonSerializer.decodeFromJsonElement<Program>(programDataObject)
                 }.mapLeft(Failure::JsonParsingException)
             }
 }
 
-// The Search API from Vier is sometimes returning non available Programs.
-// A redirect is triggered to a fixed location
-// Model this into a soft error
-internal class ProgramResponseValidator {
-    private companion object {
-        private const val NO_LONGER_AVAILABLE_REDIRECT_LOCATION = "$siteUrl/programma-niet-meer-beschikbaar"
-    }
-
-    internal suspend fun validateResponse(response: Response): Either<Failure, String> =
-        if (response.priorResponse?.headers("Location")?.firstOrNull() == NO_LONGER_AVAILABLE_REDIRECT_LOCATION) {
-            Failure.Content.ProgramNoLongerAvailable.left()
-        } else {
-            response.safeBodyString()
-        }
-}
-
 public sealed interface ProgramRepo {
-
     public suspend fun fetchPrograms(): Either<Failure, Success.Content.Programs>
-
     public suspend fun fetchProgram(programSearchKey: SearchHit.Source.SearchKey.Program): Either<Failure, Success.Content.SingleProgram>
-
 }
 
 internal class HttpProgramRepo(
-    private val client: OkHttpClient,
-    private val htmlProgramParser: HtmlProgramParser,
-    private val htmlFullProgramParser: HtmlFullProgramParser,
-    private val programResponseValidator: ProgramResponseValidator,
+    private val client: HttpClient = ktorClient,
+    private val htmlProgramParser: HtmlProgramParser = HtmlProgramParser(),
 ) : ProgramRepo {
 
-    // Scrapes the https://www.goplay.be/programmas searching for all available Programs and the details associated with it.
-    // Fortunately for us the details of a Program is encoded in the HTML DOM as a JSON
+    // Scrapes the https://www.goplay.be/programmas HTML content
     // curl -X GET "https://www.goplay.be/programmas"
-    override suspend fun fetchPrograms(): Either<Failure, Success.Content.Programs> = withContext(Dispatchers.IO) {
-        either {
-            val html = client.executeAsync(Request.Builder().get().url("$siteUrl/programmas").build()).safeBodyString().bind()
-            val programs = htmlProgramParser.parse(html).bind()
-            Success.Content.Programs(programs)
+    override suspend fun fetchPrograms(): Either<Failure, Success.Content.Programs> =
+        withContext(Dispatchers.IO) {
+            either {
+                val html = client.get<HttpResponse>("$siteUrl/programmas").readText()
```

Comment:
I think we might want to use a different way of reading the HTML to receive an unescaped version of it.